### PR TITLE
Added powerscale storage to karavictl

### PR DESCRIPTION
**Pull request by LJ-Software**
# Description

Adds support for adding PowerScale as a storage system with karavictl

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|     #107      |

# Checklist:

- [x] I have performed a self-review of my own changes.
```
$ ./bin/karavictl storage create --endpoint https://<OneFS IP> --insecure --system-id purpsefully-wrong --type powerscale --user admin --password <OMITTED>
cluster name "purpsefully-wrong" not found
$ ./bin/karavictl storage create --endpoint https://<OneFS IP> --insecure --system-id <system-id> --type powerscale --user admin --password <OMITTED>
$ ./bin/karavictl storage list
{
  "storage": {
    "powermax": {},
    "powerscale": {
      "lglw6195": {
        "Endpoint": "https://<ip-address>",
        "Insecure": true,
        "Password": "(omitted)",
        "User": "admin"
      }
    }
  }
}
```

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@LJ-Software: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

File: cmd/karavictl/cmd/testdata/onefs_api_types_System_instances_testing.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                "local_serial": "XXXXX-XXXXXX-XXXX",
+                "name": "abcd1234",
+                "onefs_version": {
+                    "build": "B_8_1_3_013(RELEASE)",
+                    "copyright": "Copyright (c) 2001-2019 EMC Corporation. All Rights Reserved.",
+                    "reldate": 1100006,
+                    "release": "v8.1.3.0",
+                    "revision": "576745869412401165",
+                    "type": "Isilon OneFS",
+                    "version": "Isilon OneFS v8.1.3.0 B_8_1_3_013(RELEASE): 0x80103500000000D:Mon Jan  7 12:10:23 PST 2019    root@sea-build10-03:/b/mnt/obj/b/mnt/src/sys/IQ.amd64.release   FreeBSD clang version 3.3 (tags/RELEASE_33/final 183502) 20130610"
```
Comment (tdawe) [inline cmd/karavictl/cmd/testdata/onefs_api_types_System_instances_testing.json:41]: `version` could be updated to just "Isilon OneFS" and remove other specific details. Similar change to set `revision`, `release`, `reldate`, `build`, and `description` (on line 2) to non-specific values.

Comment (LJ-Software) [reply cmd/karavictl/cmd/testdata/onefs_api_types_System_instances_testing.json]: Updated

File: cmd/karavictl/cmd/storage_create.go [hunk 2]
```diff
@@ -304,6 +307,34 @@ func NewStorageCreateCmd() *cobra.Command {
 					createStorageFunc(storageID)
 				}
 
+			case powerscale:
+				tempStorage = storage[powerscale]
+				if tempStorage == nil {
+					tempStorage = make(map[string]System)
+				}
+
+				psClient, err := pscale.NewClientWithArgs(context.Background(), epURL.String(), input.Insecure, 1, input.User, "Administrators", input.Password, "")
```
Comment (tdawe) [inline cmd/karavictl/cmd/storage_create.go:316]: Should "Administrators" be an input parameter instead of being hardcoded?

**Review by tdawe — APPROVED** [review]

**Review by hoppea2 — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -4,6 +4,7 @@ go 1.16
 
 require (
 	github.com/alicebob/miniredis/v2 v2.14.3
+	github.com/dell/goisilon v1.4.0 // indirect
 	github.com/dell/gopowermax v1.4.0
 	github.com/dell/goscaleio v1.2.0
```
Comment (hoppea2) [inline go.mod:9]: Do you know why goscaleio 1.2.0 is used and not 1.3.0?

File: cmd/karavictl/cmd/storage_create.go (hunk 2 above)
Comment (LJ-Software) [reply cmd/karavictl/cmd/storage_create.go:316]: `Administrators` is a default, un-deletable group on the PowerScale. AFAIK the only account that should be used `karavictl storage add` is an account that has administrator privileges on the PowerScale.

@alexemc do you think that hardcoding the `administrator` group for the PowerScale client is appropriate here?

**Pull request merged**